src/CoreWCF.BuildTools/src/OperationParameterInjectionGenerator.cs

```diff
@@ -0,0 +1,235 @@
+            }
+
+            foreach (var argument in serviceBehaviorAttribute.NamedArguments)
+            {
+                if (argument.Key == "InstanceContextMode" && !argument.Value.IsNull)
+                {
+                    return argument.Value.Value.ToString() == "2";
+                }
+            }
+
+            return false;
+        }
+
+        private static void GenerateOperationContractImplementation(GeneratorExecutionContext context, INamedTypeSymbol service, INamedTypeSymbol contract, IMethodSymbol operationContractImplementation, IMethodSymbol operationContract)
+        {
+            var SSMServiceBehaviorSymbol = context.Compilation.GetTypeByMetadataName("System.ServiceModel.ServiceBehaviorAttribute");
+            var CoreWCFServiceBehaviorSymbol = context.Compilation.GetTypeByMetadataName("CoreWCF.ServiceBehaviorAttribute");
+
+            string fileName = $"{contract.Name}_{operationContract.Name}.cs";
+            var dependencies = operationContractImplementation.Parameters.Where(x => !operationContract.Parameters.Any(p =>
+                   p.IsMatchingParameter(x))).ToArray();
+
+            bool isInstanceContextModeSingle = GetInstanceContextMode(context, service);
+            
+            var builder = new StringBuilder();
+            builder.Append($@"
+namespace {service.ContainingNamespace}
+{{
+    public partial class {service.Name}
+    {{
+");
+            builder.Append($@"        public {GetReturnType()} {operationContract.Name}({GetParameters()})
+        {{
+");
+            builder.Append($@"            System.IServiceProvider serviceProvider = CoreWCF.OperationContext.Current.InstanceContext.Extensions.Find<System.IServiceProvider>();
+            if (serviceProvider == null) throw new System.InvalidOperationException(""Missing IServiceProvider in InstanceContext extensions"");
+");
+            if (isInstanceContextModeSingle)
+            {
+                builder.Append($@"            using (var scope = serviceProvider.CreateScope())
+            {{
+");
+            }    
+            
+            AddDependenciesResolution();
+            AddMethodCall();
+
+            if (isInstanceContextModeSingle)
+            {
+                builder.Append($@"
+            }}");
+            }
+
+            builder.Append($@"
+        }}
+    }}
+}}
+");
+            context.AddSource(fileName, SourceText.From(builder.ToString(), Encoding.UTF8, SourceHashAlgorithm.Sha256));
+
+            string GetReturnType()
+                => operationContract.ReturnsVoid ?
+                    "void"
+                    : $"{operationContract.ReturnType.ContainingNamespace}.{operationContract.ReturnType.Name}";
+
+            string GetParameters()
+                => string.Join(", ", operationContract.Parameters
+                    .Select(p => $"{p.Type.ContainingNamespace}.{p.Type.Name} {p.Name}"));
+            
+            void AddDependenciesResolution()
+            {
```

Comment:
I took a look at the aspnetcore implementation around this and discovered they support a few other things which aren't available from the IServiceProvider but that we might want to add support for. The code is [here](https://github.com/dotnet/aspnetcore/blob/f58f9dcf1c10345a4fa44bbeea4777efd3b0d407/src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs#L242) and they have additional support for `HttpContext`, `HttpRequest`, `HttpResponse`, `ClaimsPrincipal`, and `CancellationToken`. While I think it's premature to do all of these (and would be feature creep), I think we can easily support HttpContext, HttpRequest and HttpResponse. The HttpContext is available as an incoming message property on the OperationContext with the key name (I think) "Microsoft.AspNetCore.Http.HttpContext". This then has the Request and Response properties to get the other two. I think it's worth adding those three. What do you think? If you want to do that as a followup item, that's fine too. I'm generally adverse to increasing scope as it makes things take longer to complete.

Reply:
Indeed let’s build a skateboard before a car. I will create an issue to trace adding the support of ˋHttpContext` and related services.